typing/jkind.ml

```diff
@@ -1610,10 +1610,18 @@ module Jkind_desc = struct
       (sub : (allowed * r) jkind_desc)
       ({ layout = lay2; mod_bounds = bounds2; with_bounds = No_with_bounds } :
         (l * allowed) jkind_desc) =
+    let axes_max_on_right =
+      (* Optimization: if the upper_bound is max on the right, then that axis is
+         irrelevant - the left will always satisfy the right along that axis. *)
```

Comment:
An even better optimization might be to set left's mod_bounds to the join of the left and right's mod_bounds. Right now this wouldn't gain us anything, but I could imagine future opimizations in `map_normalize` being able to take advantage of it. But I think this is good to leave for now.